### PR TITLE
Fix broken tests in hip ^cmake@3.21:

### PR DESCRIPTION
**Pull request by haampie**
Ping @srekolam

### CODE REVIEW

**Comment by srekolam** [comment]
@haampie , i tried the below inside hip and i did not see any error but when building rocblas i got a error with HIP_RUNTIME.it appeared to be referring to *cuda* . also it was showing RocmClang. 

amdgpu_targets = (
        'gfx900', 'gfx906', 'gfx908', 'gfx1010',
        'gfx1011', 'gfx1012'
    )
variant('amdgpu_target', default='gfx900,gfx906,gfx908', multi=True, values=amdgpu_targets)

inside cmake_args() 
archs = ",".join(self.spec.variants['amdgpu_target'].value)
args.append("-DCMAKE_HIP_ARCHITECTURES:STRING={0}".format(archs))

**Comment by haampie** [comment]
Yeah, I've added conflicts for those: #25253 see the reference there... at the moment nothing in the rocm ecosystem builds with cmake 3.21 it seems.

File: var/spack/repos/builtin/packages/hip/package.py [hunk 1]
```diff
@@ -63,10 +63,17 @@ class Hip(CMakePackage):
...
+
     # See https://github.com/ROCm-Developer-Tools/HIP/pull/2219
     patch('0004-Drop-clang-rt-builtins-linking-on-hip-host.3.7.0.patch', when='@3.7.0:3.9.0')
     patch('0004-Drop-clang-rt-builtins-linking-on-hip-host.3.10.0.patch', when='@3.10.0:4.1.0')
 
+    # Tests are broken when using cmake 3.21
+    patch('0005-Disable-tests-3.5.0.patch', when='@3.5.0       ^cmake@3.21:')
+    patch('0005-Disable-tests-3.6.0.patch', when='@3.6.0:3.8.0 ^cmake@3.21:')
+    patch('0005-Disable-tests-3.9.0.patch', when='@3.9.0:4.0.0 ^cmake@3.21:')
+    patch('0005-Disable-tests-4.1.0.patch', when='@4.1.0:      ^cmake@3.21:')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hip/package.py]: Here you can use:
```python
with when('^cmake@3.21:'):
    patch('0005-Disable-tests-3.5.0.patch', when='@3.5.0')
    ...
```
?

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/hip/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/hip/package.py]: yeah, that looks kinda nice, thanks!

**Review by alalazo — APPROVED** [review]

**Pull request merged**